src/Compilers/CSharp/Test/Emit2/Semantics/CollectionLiteralTests.cs

```diff
@@ -5644,6 +5667,2548 @@ private static void VerifyTypes(SemanticModel model, ExpressionSyntax expr, stri
+            string sourceAttribute = """
+                namespace System.Runtime.CompilerServices
+                {
+                    [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                    public sealed class CollectionBuilderAttribute : Attribute
+                    {
+                        public CollectionBuilderAttribute(Type builderType, string methodName) { }
+                    }
+                }
+                """;
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyCollectionBuilder1), "Create1")]
+                [CollectionBuilder(typeof(MyCollectionBuilder2), "Create2")]
+                public sealed class MyCollection<T> : IEnumerable<T>
+                {
+                    private readonly List<T> _list;
+                    public MyCollection(List<T> list) { _list = list; }
+                    public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
+                    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                }
+                public struct MyCollectionBuilder1
+                {
+                    public static MyCollection<T> Create1<T>(ReadOnlySpan<T> items)
+                        => new MyCollection<T>(new List<T>(items.ToArray()));
+                }
+                public struct MyCollectionBuilder2
+                {
+                    public static MyCollection<T> Create2<T>(ReadOnlySpan<T> items)
+                        => throw null;
+                }
+                """;
+            var comp = CreateCompilation(new[] { sourceAttribute, sourceA }, targetFramework: TargetFramework.Net70);
+            var refA = AsReference(comp, useCompilationReference);
+
+            string sourceB = """
+                class Program
+                {
+                    static MyCollection<int> F() => [1, 2, 3];
+                    static void Main()
+                    {
+                        F().Report();
+                    }
+                }
+                """;
+            var verifier = CompileAndVerify(new[] { sourceB, s_collectionExtensions }, references: new[] { refA }, targetFramework: TargetFramework.Net70, expectedOutput: "[1, 2, 3], ");
+            comp = (CSharpCompilation)verifier.Compilation;
+
+            var collectionType = (NamedTypeSymbol)comp.GetMember<MethodSymbol>("Program.F").ReturnType;
+            Assert.Equal("MyCollection<System.Int32>", collectionType.ToTestDisplayString());
+            TypeSymbol builderType;
+            string methodName;
+            Assert.True(collectionType.HasCollectionBuilderAttribute(out builderType, out methodName));
+            Assert.Equal("MyCollectionBuilder1", builderType.ToTestDisplayString());
+            Assert.Equal("Create1", methodName);
```

Comment:
somewhat depends on consistent ordering of attributes.  but that seems fine.  this shoudln't be allowed in practice right, since the runtime will not have AllowMultiple? (so this is more about bad actors in the eccosystem?)

Reply:
Right, this shouldn't be an issue in practice since the BCL attribute definition does not allow multiple.